packages/astro/src/vite-plugin-build-html/index.ts

```diff
@@ -253,17 +259,43 @@ export function rollupPluginAstroBuildHTML(options: PluginOptions): VitePlugin {
         assetIdMap.set(assetPath, referenceId);
       }
 
+      // Sort CSS in order of appearance in HTML (pageStyleImportOrder)
+      // This is the “global ordering” used below
+      const sortedCSSChunks = [...pureCSSChunks];
+      sortedCSSChunks.sort((a, b) => {
+        let aIndex = Math.min(
+          ...Object.keys(a.modules).map((id) => {
+            const i = pageStyleImportOrder.findIndex((url) => id.endsWith(url));
```

Comment:
Note: `id.endsWith(url)` may be a bit brittle, but fortunately we’re only sorting. If we can’t figure out the order of a module, we don’t remove it; we just keep it in whatever order we found it.